contributors/design-proposals/api-proposal-design-template.md

```diff
@@ -0,0 +1,143 @@
+# Description of your API
+
+Initial API version: v1alpha1
+
+## Owners
+
+SIG owner: list of SIG that will own the design, implementation, and support
+Individual owners: list of individual GitHub users that will own the design, implementation, and support
+
+## Abstract
+
+4-8 sentence description of the problem and solution.  Bullet points welcome.
+
+## Motivation
+
+4-8 sentence of the current state and why it is insufficient.
+
+## Use cases
+
+Bullet point list of use cases this proposal is meant to address.
+
+- As a cluster administrator...
+- As a application developer...
+- As a Kubernetes extension developer...
+
+## Requirements
+
+Bullet point list of new API requirements.  Requirements will in part be driven by use
+cases, but may have additional engineering or interoperability requirements:
+
+- User requirement
+- Engineering requirement
+- Interoperability requirement
+
+## Overview subsequent versions
+
+Subsequent versions will require their own API proposals, however it is important to include
+the plans for them here to provide a broader picture.
+
+### Beta
+
+*Use cases*
+
+- Use cases for transitioning to Beta
+
+*Requirements*
+
+- Additional requirements for transitioning to Beta
+
+### GA
+
+*Use cases*
+
+- Use cases for transitioning to Beta
+
+*Requirements*
+
+- Additional requirements for transitioning to Beta
+
+## Dependencies and deadlines
+
+List of features that will depend on these changes.
+
+List of features that these changes will depend upon.
+
+**Include any deadlines that are related.** e.g. another feature depends on this and it is slated for 1.x.
+
+## Proposed changes for new types or existing types
+
+Declare the proposed Group / Version / Kind for any new types created as part of this proposal.
+
+Declare any new fields for existing types and their version.  Will the fields be added as first class fields or
+as annotations?
+
+Describe how any new types or fields will be used by the end user.  Include sample yaml config and full
+walkthrough examples of 1 or more usecases.
+
+### Defaulting
+
+Describe defaulting that will be done
```

Comment:
This should be covered by the comments in the PR adding the types, no? Same with validation below. I think we shouldn't make people describe in prose things that are best written in the documentation. (so, checking these things should be in the instructions for api reviewers.)